src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/AddUsingsCodeActionProviderHelper.cs

```diff
@@ -38,7 +39,7 @@ public static async Task<TextEdit[]> GetUsingStatementEditsAsync(RazorCodeDocume
         var oldUsings = await FindUsingDirectiveStringsAsync(originalCSharpText, cancellationToken).ConfigureAwait(false);
         var newUsings = await FindUsingDirectiveStringsAsync(changedCSharpText, cancellationToken).ConfigureAwait(false);
 
-        var edits = new List<TextEdit>();
+        using var _ = ListPool<TextEdit>.GetPooledObject(out var edits);
```

Comment:
I'm unsure which would be better here, `ListPool` or `ArrayBuilderPool`

Reply:
To be read in the least authoratative voice you can imagine, but I've always been of the opinion that if something is never realized as an immutable collection, then using an immutable collection builder is a bit silly, and List is fine. I think they work the same way, but my hope is that List, being the more common type, might get some tiny jit advantage.

Reply:
To be fair, you're completley right. `ListPool<T>` _is_ fine when you're just going to call `ToArray()`. However, if it's likely to be a small array, `PooledArrayBuilder<T>` we'll be better because it won't even hit the pool.

Reply:
I'm much happier when there is one specific guidance we can give. So "just use pooled array builder, and call builder.toimmutable" is an easy rule to remember.

Reply:
What I've arrived at for this change 

* Default to `PooledArrayBuilder<T>` for places that are just building them and then `ToArray/ToImmutable`
* Use `ArrayBuilderPool<T>`  when there's lots of functions passing around the list but one cental start/end of the chain to manage lifetime and are _synchronous_ (use `AsRef`)
* Use `ListPool<T>` for the async version of above

Reply:
In the last and most nit-picky way, I just find it more annoying to write `ImmutableArray<T>.Builder` as the type. If there's a perf benefit I'm more than happy to amend my thinking, otherwise I'm lazy and like typing less

Reply:
Or better put, I should have merged the first two dots

* `PooledArrayBuilder<T>` for all cases, and if passed sync pass by ref. 
* `ListPool<T>` for passing to async methods